### PR TITLE
Create expiration cache

### PR DESCRIPTION
**Pull request by bpitman**

### CODE REVIEW

File: src/uid/UniqueId.java [hunk 1]
```diff
@@ -1394,4 +1401,117 @@ public static UniqueIdType stringToUniqueIdType(final String type) {
...
+            LOG.debug("executing cleanup for " + name + " [" + prev + "]");
+            int s = caches.get(prev).size();
+            caches.get(prev).clear();
+            LOG.debug("cleanup complete for " + name + " [" + prev + "] removed " + s + " items");
+          } catch (Exception e) {
+            LOG.warn("cleanup task failed", e);
+          }
+        }
+      };
+      LOG.debug("Starting background cleanup for " + name);
```
Comment (cvpatel) [inline src/uid/UniqueId.java:1447]: This is like once a day. can we make it non-debug?

File: src/utils/Config.java [hunk 2]
```diff
@@ -105,6 +105,9 @@
   /** tsd.core.cache.enable_auto_reciprocal */
   private boolean auto_reciprocal = true;
 
+  /** tsd.core.cache.expiration_age */
+  private int cache_expiration_age = 86400000; // 1 day
```
Comment (cvpatel) [inline src/utils/Config.java:109]: if all the tsdbs empty the caches roughly at the same time do we forsee any issues?

File: src/uid/UniqueId.java [hunk 3]
```diff
@@ -1394,4 +1401,117 @@ public static UniqueIdType stringToUniqueIdType(final String type) {
...
+    synchronized void clear() {
+      for (int i = 0; i < caches.size(); i++) {
+        caches.get((ptr + i) % caches.size()).clear();
+      }
+    }
+
+    U get(T key) {
+      U value = caches.get(ptr).get(key);
+      if (value == null) {
+        value = getSync(key);;
```
Comment (cvpatel) [inline src/uid/UniqueId.java:1466]: double semi-colon

File: src/utils/Config.java (hunk 2 above)
Comment (bpitman) [reply src/utils/Config.java:109]: Only entries this old that haven't been accessed more recently are deleted.  I don't think a perfect storm would be any worse than a rolling restart.

File: src/uid/UniqueId.java (hunk 1 above)
Comment (bpitman) [reply src/uid/UniqueId.java:1447]: This one only occurs once during start up.  You can also see whether or not the thread exists.  I'll change the result of the cleanup to info.

**Pull request merged**